### DISCUSSION TITLE
Map of Norway

### DISCUSSION THREAD

**Discussion by rojoaguilar24** [vega/altair#3443] [Q&A] [answered]
Hi,

first, I must say that I am rather inexperienced at programming, so, perhaps this question is very simple to some of you.

I am trying to create a combined map where I want to use a map of Norway as base map and then I would add some data that I am processing on top of that.

Thus far I have used the following code as basis 

background = alt.topo_feature(data.world_110m.url, 'countries')

# Create a base Altair map
base_map = alt.Chart(background).mark_geoshape(
    fill='lightgray',
    stroke='white'
).properties(
    width=800,
    height=500
).project('naturalEarth1')

However, it does show the entire world map and not just Norway.

So my question is if someone could help me get just Norway as base map and not the entire World map?

Thanks a lot in advance.
Daniel

**Comment by mattijn** [comment, marked as answer]
Have a look to the geoshape docs here: https://altair-viz.github.io/user_guide/marks/geoshape.html 

And parts of this blog on 'Engaging geovisualisations with Vega-Altair' is also useful: https://mattijn.github.io/posts/2023-03-07-talk-geopython-2023/.

Example for Norway:
```python
import altair as alt
import pandas as pd
import geopandas as gpd

url = "https://naciscdn.org/naturalearth/110m/cultural/ne_110m_admin_0_countries.zip"
gdf = gpd.read_file(url)  # zipped shapefile
gdf = gdf[["NAME", "CONTINENT", "POP_EST", 'geometry']]

# gdf.head()
# gdf_NO = gdf[gdf.NAME == 'Norway']

extent_roi = gdf.query("NAME == 'Norway'")
xmin, ymin, xmax, ymax = extent_roi.total_bounds
```
```python
# fit object should be a GeoJSON-like Feature or FeatureCollection
extent_roi_feature = {
    "type": "Feature",
    "geometry": {"type": "Polygon",
                 "coordinates": [[
                     [xmax, ymax],
                     [xmax, ymin],
                     [xmin, ymin],
                     [xmin, ymax],
                     [xmax, ymax]]]},
    "properties": {}
}

chart_base = alt.Chart(gdf).mark_geoshape(clip=True, fill='lightgray', stroke='white', strokeWidth=0.5).project(
    fit=extent_roi_feature,
    type='equalEarth'
)
chart_base
```
![image](https://github.com/vega/altair/assets/5186265/e9c70af0-e9b3-486e-8f19-651b69a6517d)

```python
df = pd.DataFrame({'city': ['Oslo'], 'lat': [59.91], 'lon': [10.75]})
gdf_pts = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.lon, df.lat), crs="EPSG:4326")
gdf_pts
```
![image](https://github.com/vega/altair/assets/5186265/206d9e80-242c-4476-9eec-e82e59ff3c24)
```python
chart_pts = alt.Chart(gdf_pts).mark_geoshape().encode(
    fill='city'
)

chart_base + chart_pts
```
![image](https://github.com/vega/altair/assets/5186265/1b9068f1-e28e-4e19-9034-d4230cf6ad1a)

```python
chart_text = alt.Chart(gdf_pts).mark_text(
    align='right', dy=-10
).encode(
    longitude="lon", latitude="lat", text="city"
)

chart_base + chart_pts + chart_text
```
![image](https://github.com/vega/altair/assets/5186265/9225653a-75cb-4a73-9af8-fa209d4d07bf)

Good luck with your project!

**Comment by rojoaguilar24** [comment]
Thanks a lot for your prompt and useful reply,
Daniel
